homeassistant/components/vicare/number.py

```diff
@@ -0,0 +1,194 @@
+"""Number for ViCare."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from contextlib import suppress
+from dataclasses import dataclass
+import logging
+
+from PyViCare.PyViCareDevice import Device as PyViCareDevice
+from PyViCare.PyViCareDeviceConfig import PyViCareDeviceConfig
+from PyViCare.PyViCareHeatingDevice import (
+    HeatingDeviceWithComponent as PyViCareHeatingDeviceWithComponent,
+)
+from PyViCare.PyViCareUtils import (
+    PyViCareInvalidDataError,
+    PyViCareNotSupportedFeatureError,
+    PyViCareRateLimitError,
+)
+import requests
+
+from homeassistant.components.number import NumberEntity, NumberEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ViCareRequiredKeysMixin
+from .const import DOMAIN, VICARE_API, VICARE_DEVICE_CONFIG
+from .entity import ViCareEntity
+from .utils import is_supported
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class ViCareNumberEntityDescription(NumberEntityDescription, ViCareRequiredKeysMixin):
+    """Describes ViCare sensor entity."""
+
+    value_setter: Callable[[PyViCareDevice, float], str | None] | None = None
+
+
+CIRCUIT_SENSORS: tuple[ViCareNumberEntityDescription, ...] = (
+    ViCareNumberEntityDescription(
+        key="heating curve shift",
+        name="Heating Curve Shift",
```

Comment:
Can we use translation keys? Not sure


```suggestion
        name="Heating curve shift",
```